pype/plugins/global/publish/extract_review.py

```diff
@@ -13,22 +19,1064 @@ class ExtractReview(pyblish.api.InstancePlugin):
+                new_repre_files.append(repr_file % frame)
+
+            new_repre["sequence_file"] = repr_file
+            full_output_path = os.path.join(
+                staging_dir, filename_base, repr_file
+            )
+
+        else:
+            repr_file = "{}_{}.{}".format(
+                filename, filename_suffix, output_ext
+            )
+            full_output_path = os.path.join(staging_dir, repr_file)
+            new_repre_files = repr_file
+
+        # Store files to representation
+        new_repre["files"] = new_repre_files
+
+        # Make sure stagingDire exists
+        staging_dir = os.path.normpath(os.path.dirname(full_output_path))
+        if not os.path.exists(staging_dir):
+            self.log.debug("Creating dir: {}".format(staging_dir))
+            os.makedirs(staging_dir)
+
+        # Store stagingDir to representaion
+        new_repre["stagingDir"] = staging_dir
+
+        # Store paths to temp data
+        temp_data["full_input_path"] = full_input_path
+        temp_data["full_input_path_single_file"] = full_input_path_single_file
+        temp_data["full_output_path"] = full_output_path
+
+        self.log.debug("Input path {}".format(full_input_path))
+        self.log.debug("Output path {}".format(full_output_path))
+
+        return full_input_path, full_output_path
+
+    def audio_args(self, instance, temp_data):
+        """Prepares FFMpeg arguments for audio inputs."""
+        audio_in_args = []
+        audio_filters = []
+        audio_out_args = []
+        audio_inputs = instance.data.get("audio")
+        if not audio_inputs:
+            return audio_in_args, audio_filters, audio_out_args
+
+        for audio in audio_inputs:
+            # NOTE modified, always was expected "frameStartFtrack" which is
+            # STRANGE?!!! There should be different key, right?
+            # TODO use different frame start!
+            offset_seconds = 0
+            frame_start_ftrack = instance.data.get("frameStartFtrack")
+            if frame_start_ftrack is not None:
+                offset_frames = frame_start_ftrack - audio["offset"]
+                offset_seconds = offset_frames / temp_data["fps"]
+
+            if offset_seconds > 0:
+                audio_in_args.append(
+                    "-ss {}".format(offset_seconds)
+                )
+            elif offset_seconds < 0:
+                audio_in_args.append(
+                    "-itsoffset {}".format(abs(offset_seconds))
+                )
+
+            audio_in_args.append("-i \"{}\"".format(audio["filename"]))
+
+        # NOTE: These were changed from input to output arguments.
+        # NOTE: value in "-ac" was hardcoded to 2, changed to audio inputs len.
+        # Need to merge audio if there are more than 1 input.
+        if len(audio_inputs) > 1:
+            audio_out_args.append("-filter_complex amerge")
+            audio_out_args.append("-ac {}".format(len(audio_inputs)))
```

Comment:
Is this ok? I've changed from `"-ac 2"` to `"-ac {}".format(len(audio_inputs))` by FFmpeg documentation https://ffmpeg.org/ffmpeg-all.html#Audio-Options

Reply:
Definitely good idea to have this procedural. Not that we are using somewhere more then to channels per input. Well is this actually reflecting 2 as channels or input of files? If the `audio_inputs` is list of files that this is false value. It should be run trough _ffprobe_ to check channels.

Reply:
`audio_args` require more changes than this. I would like to make those changes in different PR since audio is not used right now, I don't know how it should work and it would take a long time to find out.